### PR TITLE
Transactions TTL

### PR DESCRIPTION
**Pull request by eadventurous**
Signed-off-by: Egor Ivkov <e.o.ivkov@gmail.com>

### CODE REVIEW

File: iroha/src/tx.rs [hunk 1]
```diff
@@ -105,6 +111,18 @@ impl AcceptedTransaction {
         hash.copy_from_slice(&vec_hash);
         hash
     }
+
+    /// If this transaction is waiting longer than specified in `transaction_ttl` or `proposed_ttl_ms`.
```
Comment (vmarkushin) [inline iroha/src/tx.rs:115]: I think this comment can be rewritten to be more specific

File: iroha/src/tx.rs [hunk 2]
```diff
@@ -25,14 +28,17 @@ struct Payload {
     /// An ordered set of instructions.
     instructions: Vec<Instruction>,
     /// Time of creation (unix time, in milliseconds).
-    creation_time: String,
+    creation_time: u64,
+    /// The transaction will be dropped after this time if it is still in a `Queue`.
+    proposed_ttl_ms: u64,
```
Comment (humb1t) [inline iroha/src/tx.rs:33]: I think this field can be named `time_to_live_ms`

File: iroha/src/queue.rs [hunk 3]
```diff
@@ -1,17 +1,20 @@
 use self::config::QueueConfiguration;
 use crate::prelude::*;
+use std::time::Duration;
 
 #[derive(Debug)]
 pub struct Queue {
     pending_tx: Vec<AcceptedTransaction>,
     maximum_transactions_in_block: usize,
+    transaction_ttl: Duration,
```
Comment (humb1t) [inline iroha/src/queue.rs:9]: I think this field can be named `transactions_time_to_live_ms`.

File: iroha/src/tx.rs (hunk 2 above)
Comment (eadventurous) [reply iroha/src/tx.rs:33]: TTL is quite a common term from network terminology, I guess it will be clear for developers/contributors?

**Pull request merged**